src/file-size/format-file-size.js

```diff
@@ -0,0 +1,43 @@
+import formatDecimal from '../number/format-decimal';
+import validateFormatValue from '../number/validate-format-value';
+
+function getFileSizeUnits(localeData) {
+	const fileSizeUnits = Object.freeze([
+		{unit: localeData.fileSize.units.gigabyte, num: Math.pow(1024, 3)},
+		{unit: localeData.fileSize.units.megabyte, num: Math.pow(1024, 2)},
+		{unit: localeData.fileSize.units.kilobyte, num: 1024},
+		{unit: localeData.fileSize.units.bytes, num: 1}
+	]);
+
+	return fileSizeUnits;
+}
+
+export default function(numBytes, localeData) {
+	numBytes = validateFormatValue(numBytes);
+
+	const fileSizeUnits = getFileSizeUnits(localeData);
+
+	let formatUnit,
+		size;
+
+	if(numBytes === 0) {
+		formatUnit = localeData.fileSize.units.bytes;
+		size = 0;
+	} else if(Math.abs(numBytes) === 1){
+		formatUnit = localeData.fileSize.units.byte;
+		size = numBytes;
+	} else {
+		for (var unitSize of fileSizeUnits) {
+			if (Math.abs(numBytes) >= unitSize.num) {
+				formatUnit = unitSize.unit;
+				size = formatDecimal(numBytes/unitSize.num, localeData.number, { maximumFractionDigits: 2 });
+				break;
+			}
+		}
+	}
+
+	let ret = localeData.fileSize.patterns.fileSizePattern;
+	ret = ret.replace('{number}', size).replace('{unit}', formatUnit);
```

Comment:
@jstefaniuk-d2l: could you use [process-pattern](https://github.com/Brightspace/intl/blob/master/src/util/process-pattern.js) here instead of the double replace? The other formatters should have examples of how to use it.